### PR TITLE
Add zap decollateralize hardhat task

### LINKED ISSUE #405: Hardhat task for deploying SolidZapDecollateralize

### PR DESCRIPTION
**Pull request by andreistefanwork**
Closes #405

### CODE REVIEW

**Review by cawabunga — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -75,6 +75,17 @@ The following tasks are available in Hardhat:
    - `--uni-proxy`: The address of the Gamma UniProxy contract used for deploying liquidity.
    - `--solid-staking`: The address of our SolidStaking contract, to stake with recipient.
 
+5. Deploy a new SolidZapDecollateralize contract:
+   ```shell
+   yarn hardhat --network localhost deploy-zap-decollateralize [OPTIONS]
+   ```
+
+   Options:
+   - `--router`: The address of the router used to perform token swaps.
```
Comment (cawabunga) [inline README.md:84]: I was thinking if router needs to be hardcoded. Can we just pass it as an argument with encoded swap?

Comment (andreistefanwork) [reply README.md:84]: what we ultimately need is to do `router.call(encodedSwap);`

we can encode the address of the router into the swap itself on the frontend, e.g. the encoded swap would have 2 parts, (address router ; bytes actualSwap)

then we would do an on-chain `abi.decode` to obtain the 2 parts, and call `router.call(actualSwap);`

but we know the router from the beginning, it's simple to have it during contract deployment as constructor argument. it also enables `IWETH(weth).approve(_router, type(uint).max);` in the constructor

Comment (cawabunga) [reply README.md:84]: What if we decide to support another router? Or allow users to decide what router to use.

Comment (andreistefanwork) [reply README.md:84]: We deploy another zap with different router

Comment (andreistefanwork) [reply README.md:84]: Cause if the router is different, we might also need to adjust what we do after `router.call(encodedSwap);`. 
Kyberswap router for example returns amount swapped from `router.call(encodedSwap)`, and we can process it

**Pull request merged**